front-end/src/visitor/VisitorFunction.cpp

```diff
@@ -20,25 +20,48 @@ antlrcpp::Any StartVisitor::visitArgumentList(MapleGrammarParser::ArgumentListCo
 antlrcpp::Any StartVisitor::visitFunctionDefinition(MapleGrammarParser::FunctionDefinitionContext *ctx) {
     const string &name = ctx->ID()->getText();
 
-    if (auto symbol = currentSymbolTable->lookup(name)) {
-        cerr << "Duplicate declaration of " << name << endl;
-        cerr << "Found : " << symbol->getDeclaration() << endl;
-        printDebugInfo(cerr, ctx);
-        throw std::runtime_error("Duplicate definition");
-    }
-
-
     Type type = Type::VOID;
 
     if (ctx->TYPE() != nullptr) {
         type = getTypeFromString(ctx->TYPE()->getText());
     }
 
-
     auto fDef = new FunctionDefinition(
             type,
             name
     );
+
+    auto params = *(vector<FunctionParam *> *) visit(ctx->typeList());
+
+    if (auto symbol = currentSymbolTable->lookup(name)) {
+        FunctionDefinition *temp = (FunctionDefinition *) symbol->getDeclaration();
```

Comment:
Do not use static cast, use a dynamic_cast to check if its a FunctionDefinition

Reply:
Done